recipe/meta.yaml

```
@@ -12,32 +12,30 @@ source:
   sha256: {{ sha256 }}
 
 build:
-  number: 1
+  number: 2
   skip: true  # [win]
 
 requirements:
   build:
     - {{ compiler('fortran') }}
     - {{ compiler('c') }}
   host:
+    - setuptools
+    - libblas
     - libcblas
     - liblapack
     - python
-    - setuptools
-    # once we hit 1.16.3 it should be fine again
-    # I read that numpy-devel should work, so...
-    - numpy <=1.15.3|>=1.16.3
-    - scipy >=0.18
-    # For cythonizing stuff
     - cython
-    - pyparsing
+    - numpy >=1.20
```

It doesn't look like dependencies actually require `numpy>=1.20`. Can you change to only `numpy` and rerender? If that doesn't work, I can have a look at the error, too.

```suggestion
    - numpy
```

---

But then I would have 2 separate branches building the same feedstock for py3.6? No?
That is what I am trying to avoid.

---

I agree, I don't have any requirements for numpy (only >=1.13). But the problem is that I build against numpy API and in numpy 1.20 the internal array has changed (see [here](https://github.com/numpy/numpy/releases/tag/v1.20.0) under C API changes.
I have had users complaining that my latest feedstock (build against 1.20) doesn't work on their setup since they have another package required pre 1.20 numpy versions... :( sigh... ;)

---

Yes, this will build for the minimal supported `numpy` version on conda-forge (For Python 3.6 this is 1.17 for now) and it should be compatible with all newer NumPy versions (The C-API of NumPy guarantees forward compatibility). 

> But then I would have 2 separate branches building the same feedstock for py3.6?

No, you should. There should be only a single CI entry.

---

But my cythonization does not insert the requested operation required to be backward compatible...
So numpy's release page says that 

> Starting with NumPy 1.20 code that is compiled against an earlier
version will be API incompatible with NumPy 1.20. The fix is to either
compile against 1.16.6 (if the NumPy 1.16 release is the oldest release
you wish to support), or manually inline the macro by replacing it with
the new definition:

and I don't have control over this now?

My problem is that users on py3.7 may have a package that requires numpy <1.20 (I don't know why ;) ) and then it will install, but when the users runs my installation, it crashes...

I am not sure exactly how you would suggest me to go forward? (but thanks a lot for your effort!!!)
should I ditch the 2 branch method?

---

That is a problem I personally reported and is fixed here in conda-forge by simply using the global pin. It is important that you don't compile with `numpy<1.16.6`. I'm not sure what you mean with the two-branch-method but please just keep using the default way.

---

Thanks! Ok, so I should do something like this:
```yaml

host:
  ...
  - numpy >=1.16.6
run:
  - numpy
```
and that's it?

---

No, please just apply my suggestion as above and rerender.

---

as for the two branch model, I was referring to [this](https://conda-forge.org/docs/maintainer/updating_pkgs.html?highlight=branch#maintaining-several-versions) which I successfully use for other feedstocks where I have multiple versions I maintain at the same time. :)

---

Now I am a bit confused, since then I would have this:
```yaml
host:
 - numpy
run:
  - {{ pin_compatible('numpy', min_pin='x.x') }}
```
but is that what I want? I don't want 1.16.6 to be the minimum version...

---

I will try only changing the above you mentioned and see.

---

> as for the two branch model, I was referring to this which I successfully use for other feedstocks where I have multiple versions I maintain at the same time. :)

That is meant for supporting multiple versions of the package, i.e. `sisl` in this case, not different dependency sets. For that you can use `conda_build_config.yaml`. That is though not needed here.

---

```
  - {{ pin_compatible('numpy', min_pin='x.x') }}
```

is fine as it will add a `>=1.17,<2` pin automatically to run. You could though omit the `min_pin='x.x'` part as this is not necessary and that default would be OK.